src/hotspot/share/gc/shenandoah/shenandoahDegeneratedGC.cpp

```diff
@@ -96,13 +96,6 @@ void ShenandoahDegenGC::op_degenerated() {
 
 #ifdef ASSERT
   if (heap->mode()->is_generational()) {
-    if (_generation->is_global()) {
-      // We can only get to a degenerated global cycle _after_ a concurrent global cycle
-      // has been cancelled. In which case, we expect the concurrent global cycle to have
-      // cancelled the old gc already.
-      assert(!heap->is_old_gc_active(), "Old GC should not be active during global cycle");
-    }
-
```

Comment:
Is this not true?

You could rearrange the test into the assertion check though:

```
    assert(!_generation->is_global() || !heap->is_old_gc_active, "Old GC should not be active during global cycle");
```
(or its de-Morganization, if you prefer).

Reply:
With this change, if the global cycle degenerates after final mark there could now be regions waiting to be coalesced and filled, so the old gc state is not idle.